client/my-sites/people/role-select/index.jsx

```diff
@@ -135,5 +138,32 @@ export default React.createClass( {
 				}
 			</FormFieldset>
 		);
+	},
+
+	renderSelect() {
+		const { selectedValue, defaultOption, selectedCount, onSelect, calculatePath } = this.props;
+
+		const options = [ defaultOption ];
+
+		forEach( this.state.roles, ( roleObject, key ) => {
+			options.push( { value: key, label: roleObject.display_name, path: calculatePath(key) } );
+		} );
```

Comment:
This could also be created immutably in a single expression.

```js
const options = [ defaultOption ]
	.concat( this.state.roles.map( ( role, key ) => ( {
		value: key,
		label: role.display_name,
		path: calculatePath( key ),
	} ) ) );
```